mobly/controllers/android_device_lib/jsonrpc_client_base.py

```diff
@@ -271,6 +260,20 @@ def _rpc(self, method, *args):
             raise ProtocolError(ProtocolError.MISMATCHED_API_ID)
         return result['result']
 
+    def _is_app_running(self):
```

Comment:
The current implementation fixes sl4a too actually.
I agree having a version check would be nice going forward. Could you file an FR?

In this particular case, since we're still active users of sl4a and you've seen this issue, I'd like to fix it too :)

I don't think we need to start changing the protocol for this particular case since it's so easy to actually check... we just need ANY output from the socket. This could be a connection attempt, an invalid message that causes it to spit out an error, anything. So I'd like to push off the pain of forking/duplicating/versioning/cherrypicking until there's no easy workaround available. :)